icc-x-api/icc-patient-x-api.ts

```diff
@@ -575,12 +575,24 @@ export class IccPatientXApi extends iccPatientApi {
   }
 
   encrypt(user: models.UserDto, pats: Array<models.PatientDto>): Promise<Array<models.PatientDto>> {
+    const fixEncryptionKeys = (p: models.PatientDto) => {
+      if (p.delegations && p.delegations[(user.healthcarePartyId || user.patientId)!])
+        return this.initEncryptionKeysForDelegates(
+          user,
+          p,
+          Object.keys(p.delegations)
+        ) as Promise<any>
+      throw new Error(
+        `Patient ${p.id} has no delegation or encryption key for hcp ${
+          user.healthcarePartyId || user.patientId
+        }`
+      )
+    }
     return Promise.all(
       pats.map((p) =>
-        (p.encryptionKeys &&
-        Object.keys(p.encryptionKeys).some((k) => !!p.encryptionKeys![k].length)
+        (p.encryptionKeys && p.encryptionKeys[(user.healthcarePartyId || user.patientId)!]?.length
```

Comment:
The information in `encryptedSelf` is lost (MS only encrypts the field `notes`). The current hcp cannot read the patient because the encryption key in the delegation has been replaced by a new one in `encryptionKeys`. Since only the hcps in the encryptionKeys can still read the patients, there is no easy way to retrieve the data.
Since no doctor has complained about lost notes yet (afaik), I think it's the most pragmatic solution.